<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
  <channel>
    <title>疑难杂症 on 月盾的博客</title>
    <link>https://www.yuedun.wang/categories/%E7%96%91%E9%9A%BE%E6%9D%82%E7%97%87/</link>
    <description>Recent content in 疑难杂症 on 月盾的博客</description>
    <generator>Hugo -- gohugo.io</generator>
    <language>zh-cn</language>
    <lastBuildDate>Mon, 15 Jun 2020 04:36:50 +0000</lastBuildDate><atom:link href="https://www.yuedun.wang/categories/%E7%96%91%E9%9A%BE%E6%9D%82%E7%97%87/index.xml" rel="self" type="application/rss+xml" />
    <item>
      <title>puppeteer TypeError: text is not iterable</title>
      <link>https://www.yuedun.wang/blogdetail/5ee6fae2e8775d3b7f2ee0c6/</link>
      <pubDate>Mon, 15 Jun 2020 04:36:50 +0000</pubDate>
      
      <guid>https://www.yuedun.wang/blogdetail/5ee6fae2e8775d3b7f2ee0c6/</guid>
      <description>(node:9828) UnhandledPromiseRejectionWarning: TypeError: text is not iterable at Keyboard.type (D:\workspace\auto-ui\node_modules\puppeteer-core\lib\Input.js:160:24) at Keyboard.&amp;lt;anonymous&amp;gt; (D:\workspace\auto-ui\node_modules\puppeteer-core\lib\helper.js:112:23) at openBrowser (D:\workspace\auto-ui\test-email\index.js:106:25) at process._tickCallback (internal/process/next_tick.js:68:7) (node:9828) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3) (node:9828) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.</description>
    </item>
    
    <item>
      <title>win10家庭版组策略</title>
      <link>https://www.yuedun.wang/blogdetail/5b8b9d63b1f3aa6dab1430d2/</link>
      <pubDate>Sun, 02 Sep 2018 08:20:51 +0000</pubDate>
      
      <guid>https://www.yuedun.wang/blogdetail/5b8b9d63b1f3aa6dab1430d2/</guid>
      <description>win10家庭版本身不支持组策略，但还是有办法让其支持。 复制下面内容到文本文件：
@echo off pushd &amp;#34;%~dp0&amp;#34; dir /b C:\Windows\servicing\Packages\Microsoft-Windows-GroupPolicy-ClientExtensions-Package~3*.mum &amp;gt;List.txt dir /b C:\Windows\servicing\Packages\Microsoft-Windows-GroupPolicy-ClientTools-Package~3*.mum &amp;gt;&amp;gt;List.txt for /f %%i in (&amp;#39;findstr /i . List.txt 2^&amp;gt;nul&amp;#39;) do dism /online /norestart /add-package:&amp;#34;C:\Windows\servicing\Packages\%%i&amp;#34; pause 后缀名保存为cmd,以管理员身份执行：
然后WIN+R中执行gpedit.msc即可打开组策略。
如此可以关闭占用CPU过高的Windows Defender。</description>
    </item>
    
    <item>
      <title>github提交不记录Contributions</title>
      <link>https://www.yuedun.wang/blogdetail/5aaf2794a55bb1fd38001c63/</link>
      <pubDate>Mon, 19 Mar 2018 02:59:32 +0000</pubDate>
      
      <guid>https://www.yuedun.wang/blogdetail/5aaf2794a55bb1fd38001c63/</guid>
      <description>github上提交了很多commit但是没有Contributions绿色方块，原因是提交的email和github不匹配，使用git log查看记录中使用的邮箱是否是github的邮箱，如果不是也不需要做什么修改，只需要在github上添加对应的邮箱地址即可。
 右上角头像-settings-emails-Add email address
 验证邮箱后就会立马重新统计</description>
    </item>
    
    <item>
      <title>请求发送到Nodejs服务器以后响应非常慢</title>
      <link>https://www.yuedun.wang/blogdetail/5a24ed6c2d700330ae669e89/</link>
      <pubDate>Mon, 04 Dec 2017 06:38:36 +0000</pubDate>
      
      <guid>https://www.yuedun.wang/blogdetail/5a24ed6c2d700330ae669e89/</guid>
      <description>在项目中遇到过这种情况：请求发送后迟迟没有响应，但也并没有报错，在代码中一步步调试都没有发现问题，明明已经走到最后返回数据一步了客户端却没有返回。原因是在中间件中使用了redis或memcache缓存，然后连接缓存服务失败，然后就会一直等待，直到连接缓存超时才会继续执行后续操作。</description>
    </item>
    
    <item>
      <title>BAE上连接mongodb每隔十多小时就不能连接的问题(二)</title>
      <link>https://www.yuedun.wang/blogdetail/5449dac894f76294f5e60fe1/</link>
      <pubDate>Fri, 24 Oct 2014 04:51:20 +0000</pubDate>
      
      <guid>https://www.yuedun.wang/blogdetail/5449dac894f76294f5e60fe1/</guid>
      <description>前段时间写了《BAE上连接mongodb每隔十多小时就不能连接的问题(一)》之后暂时的解决了连不上的问题，每隔十小时重启一次，但是这个方法却没有彻底解决问题，偶尔还会出现三四小时就连不上，实在搞不懂问题到底出在哪，到底是bae的mongodb的问题还是mongoose中间件的问题，现象是有做open操作，但是却没有open事件发出，那么我想是不是mongoose存在bug，翻看了源码也没看出来个所以然，不过大概是觉得要重新打开需要保证连接已经关闭的，那么干脆在监听到error事件时就将状态直接改为disconnected，反正是要调用db.close()方法进行关闭连接的，可能close()方法不好使，没有完全关闭，如果我手动将状态设为disconnected，close方法中也会判断是否是这个状态，如果是就直接返回，省的多走其他步骤了。不过这样一来就不会有close事件发出了，根据我所写代码的逻辑，那就不会调用open()方法了，但实际情况确实程序可以正常运行，说明已经重连上了，原来在代码中添加了这个属性：
Js代码 :
var opts = { db: { native_parser: true }, server: { poolSize:4, auto_reconnect: true }, user: username, pass: password }; 上面的红色字体，只能说是可能这个参数起作用了。
不过由此看来，auto_reconect这个参数要起作用必须是在连接断开的情况下，说明close()方法有时候并没有完全断开连接。</description>
    </item>
    
  </channel>
</rss>
